src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -173,6 +175,17 @@ def check_register_args(
         )
 
 
+def check_execute_as_arg(execute_as: typing.Literal["caller", "owner"]):
+    if (
+        not isinstance(execute_as, str)
+        or execute_as.lower() not in EXECUTE_AS_WHITELIST
+    ):
+        raise TypeError(
+            f"'execute_as' value '{execute_as}' is invalid, choose from "
+            f"{', '.join(EXECUTE_AS_WHITELIST, )}"
+        )
```

Comment:
You communicate in your type-hint that it only accepts either `"caller"`, or `"owner"` as input and then you check whether it's something else.
Think of type-hints as a contract, you the writer of the function promise that the function works for these inputs and gives this type back.
If I take your type-hints into account the check itself makes no sense. I'd like if we removed this altogether.

Reply:
This check was already present in the old code for checking `execute_as` args. Now if it is removed, is that a behavior change? Also, I believe the old code should've raise ValueError instead of TypeError